src/test/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMDescriptorTest.java

```diff
@@ -34,25 +61,232 @@
+    }
+
+    @Test
+    public void testDoFillProjectNameItemsServerNonexistent() {
+        HttpResponses.HttpResponseException response = (HttpResponses.HttpResponseException) descriptor.doFillProjectNameItems("non-existent", null, "test");
+        assertEquals("java.lang.Exception: The provided Bitbucket Server serverId does not exist", response.getMessage());
+    }
+
+    @Test
+    public void testDoFillRepositoryNameItemsBitbucketClientException() {
+        String searchTerm = "test";
+        String myProject = "myProject";
+        BitbucketRepositorySearchClient client = mock(BitbucketRepositorySearchClient.class);
+        when(client.get(searchTerm)).thenThrow(new BitbucketClientException("Bitbucket had an exception", 400, "Some error from Bitbucket"));
+        when(bitbucketClientFactory.getRepositorySearchClient(myProject)).thenReturn(client);
+        HttpResponses.HttpResponseException response = (HttpResponses.HttpResponseException) descriptor.doFillRepositoryNameItems(SERVER_ID_VALID, null, myProject, searchTerm);
+        assertEquals("com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException: - response: 400", response.getMessage());
+    }
+
+    @Test
+    public void testDoFillRepositoryNameItemsCredentialsIdBlank() {
+        String searchTerm = "test";
+        HttpResponse response = descriptor.doFillRepositoryNameItems(SERVER_ID_VALID, "", "myProject", searchTerm);
+        verifyRepositorySearchResponse(searchTerm, "myProject", response);
+    }
+
+    @Test
+    public void testDoFillRepositoryNameItemsCredentialsIdNull() {
+        String searchTerm = "test";
+        String projectName = "myProject";
+        HttpResponse response = descriptor.doFillRepositoryNameItems(SERVER_ID_VALID, null, projectName, searchTerm);
+        verifyRepositorySearchResponse(searchTerm, projectName, response);
+    }
+
+    @Test
+    public void testDoFillRepositoryNameItemsCustomCredentials() throws Exception {
+        String credentialId = UUID.randomUUID().toString();
+        CredentialsStore store = CredentialsProvider.lookupStores(bbJenkins).iterator().next();
+        Domain domain = Domain.global();
+        Credentials credentials = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, credentialId,
+                "", "myUsername", "myPassword");
+        store.addCredentials(domain, credentials);
+
+        String searchTerm = "test";
+        HttpResponse response = descriptor.doFillRepositoryNameItems(SERVER_ID_VALID, credentialId, "myProject", searchTerm);
+        verifyRepositorySearchResponse(searchTerm, "myProject", response);
+    }
+
+    @Test
+    public void testDoFillRepositoryNameItemsProjectNameBlank() {
+        HttpResponses.HttpResponseException response = (HttpResponses.HttpResponseException) descriptor.doFillRepositoryNameItems(SERVER_ID_VALID, null, "", "test");
+        assertEquals("java.lang.Exception: The projectName must be present", response.getMessage());
```

Comment:
As you may have seen this doesn't work any more :( 
Since we're correctly returning without a stack we don't get the `getMessage` so I think we need to be a bit more verbose in how this is done. 
```suggestion
        StaplerResponse resp = mock(StaplerResponse.class);
        response.generateResponse(null, resp, null);
        verify(resp).sendError(eq(400), eq("The projectName must be present"));response.getMessage());
```
That should do the trick, but you may want to take part of it and reuse mocks and whatnot.